lib/stdlib/src/beam_lib.erl

```diff
@@ -906,19 +906,15 @@ beam_filename(File) ->
     filename:rootname(File, ".beam") ++ ".beam".
 
 
-uncompress(Binary0) ->
-    {ok, Fd} = ram_file:open(Binary0, [write, binary]),
-    {ok, _} = ram_file:uncompress(Fd),
-    {ok, Binary} = ram_file:get_file(Fd),
-    ok = ram_file:close(Fd),
-    Binary.
-
-compress(Binary0) ->
-    {ok, Fd} = ram_file:open(Binary0, [write, binary]),
-    {ok, _} = ram_file:compress(Fd),
-    {ok, Binary} = ram_file:get_file(Fd),
-    ok = ram_file:close(Fd),
-    Binary.
+uncompress(Binary) ->
+    try
+	zlib:gunzip(Binary)
+    catch
+	_:_ -> Binary
```

Comment:
Maybe this should just fail if an uncompressed file is given but for this pull request I decided to preserve the previous behaviour which would simply return the binary.